Explain automatic checkout completion option (#1334)
saleor/saleor-docs: docs/developer/payments/transactions.mdx

Follow [Transactions Webhook Events](/developer/extending/webhooks/synchronous-events/transaction.mdx) guide.

### Automatic checkout completion

The checkout process can be automatically completed based on the transaction events, once the full payment is made.

> A checkout is considered fully paid when the [`checkout.charge_status`](/api-reference/checkout/objects/checkout#checkoutchargestatuscheckoutchargestatusenum---) 

Reviewer: ```suggestion
A checkout is considered fully paid when the [`Checkout.chargeStatus`](/api-reference/checkout/objects/checkout#checkoutchargestatuscheckoutchargestatusenum---) 
```

> A checkout is considered fully paid when the [`checkout.charge_status`](/api-reference/checkout/objects/checkout#checkoutchargestatuscheckoutchargestatusenum---) 

Reviewer: If this is the case, then I think our logic is incorrect: if channel uses `transactionFlowStrategy.AUTHORIZATION`, then "paid" = `authorizeStatus.FULL`

is set to `FULL`.
This occurs when the total of the [`chargedAmount`](/api-reference/payments/objects/transaction-item#transactionitemchargedamountmoney---) 
and [`chargePendingAmount`](/api-reference/payments/objects/transaction-item#transactionitemchargependingamountmoney---)
from all transactions covers the [`checkout.totalPrice`](/api-reference/checkout/objects/checkout#checkoutsubtotalpricetaxedmoney---).

To enable automatic checkout completion, update the channel checkout settings using the configuration example below:
```graphql
mutation UpdateChannel($id: ID!,$input: ChannelUpdateInput!){
  channelUpdate(id: $id, input: $input){
    channel{
      id
      checkoutSettings {
        automaticallyCompleteFullyPaidCheckouts
      }
    }
    errors{
      field
      code
      message
    }
  }
}
```
Query variables:
```json
{
  "id": "Q2hhbm5lbDox",
  "input": {
    "checkoutSettings": {
      "automaticallyCompleteFullyPaidCheckouts": true
    }
  }
}
```

:::info
Enabling automatic checkout completion will not disrupt storefront functionality.

> If `checkoutComplete` is called for a checkout that has already been converted to an order,

Reviewer: ```suggestion
If `checkoutComplete` is called for a `Checkout` that has already been converted into an `Order`,
```

the corresponding `Order` instance will be return.
:::

:::warning

> The checkout will be automatically completed even if some authorized or charged amounts are still pending,

Reviewer: ```suggestion
Checkout will be automatically completed even if some authorized or charged amounts are still pending,
```

as long as the total amount is covered (`authorizeStatus` will be `FULL`).
However, once the checkout is converted to an `Order`, the `authorizeStatus` will be `NONE`

> because pending amounts are not considered in the order status calculation.

Reviewer: ```suggestion
because pending amounts are not considered in the order status calculation.

Keep in mind that a pending Transaction might be rejected later.
```
